### PR TITLE
feat: Add Instagram URL rewriter to kkinstagram.com

### PR DESCRIPTION
**Pull request by privacydied**
Implements a new Instagram handler that rewrites instagram.com URLs to kkinstagram.com for improved Discord embeds on Posts and Reels.

Changes:
- Add Instagram.cs handler with URL pattern matching and rewriting
- Support instagram.com, www.instagram.com, m.instagram.com, l.instagram.com hosts
- Handle /p/ (posts), /reel/, and /reels/ paths
- Preserve query parameters and URL fragments during rewrite
- Register Instagram handler in Program.cs DI container
- Update README with Instagram support documentation

Implementation details:
- Zero external API calls (pure URL transformation)
- Idempotent: skips already-rewritten kkinstagram.com URLs
- Security: validates exact host match to prevent spoofing
- Follows existing handler patterns (similar to Reddit.cs)
- Can be disabled via Bot:DisabledSites configuration

Example: https://instagram.com/p/ABC123/ → https://kkinstagram.com/p/ABC123/
	new file:   INSTAGRAM_IMPLEMENTATION_SUMMARY.md
	modified:   README.md
	new file:   SaucyBot.Tests/Unit/Site/InstagramTest.cs
	modified:   SaucyBot/Program.cs
	new file:   SaucyBot/Site/Instagram.cs

### CODE REVIEW

File: SaucyBot/Site/Instagram.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    {
+        var originalUrl = match.Value;
+        var host = match.Groups["host"].Value.ToLowerInvariant();
+
+        // Idempotent: skip if already rewritten
+        if (host.Contains("kkinstagram.com"))
+        {
+            _logger.LogDebug("Instagram URL already rewritten to kkinstagram.com, skipping: {Url}", originalUrl);
+            return null;
+        }
```
Comment (Sn0wCrack) [inline SaucyBot/Site/Instagram.cs]: The `host` group of the regex match will never contain "kkinstagram.com" as the regex cannot match against it: https://regex101.com/r/tQqzpB/1

Additionally even if it did, this is made redundant by your `IsValidInstagramHost` call below.

File: SaucyBot/Site/Instagram.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+        // Security: ensure exact host match (not subdomains like instagram.com.evil.tld)
+        if (!IsValidInstagramHost(host))
+        {
+            _logger.LogDebug("Invalid Instagram host detected, skipping: {Host}", host);
+            return null;
+        }
```
Comment (Sn0wCrack) [inline SaucyBot/Site/Instagram.cs]: Similar to above, the regex will never capture anything other than www.instagram.com, instagram.com, m.instagram.com or l.instagram.com. So I think this check is redundant.

File: SaucyBot/Site/Instagram.cs [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        var path = match.Groups["path"].Value;
+        var query = match.Groups["query"].Success ? match.Groups["query"].Value : string.Empty;
+        var fragment = match.Groups["fragment"].Success ? match.Groups["fragment"].Value : string.Empty;
+
+        // Check if path is in accepted list
+        if (!IsAcceptedPath(path))
+        {
+            _logger.LogDebug("Instagram path not in accepted list (posts/reels), skipping: {Path}", path);
+            return null;
+        }
```
Comment (Sn0wCrack) [inline SaucyBot/Site/Instagram.cs]: This is also checked in your regex by explicitly matching against p, reel and reels at the start of the path portion of your match.

File: INSTAGRAM_IMPLEMENTATION_SUMMARY.md
Comment (Sn0wCrack) [inline INSTAGRAM_IMPLEMENTATION_SUMMARY.md:1]: I'm not really sure what the point of this empty markdown file is?

**Review by Sn0wCrack — CHANGES_REQUESTED** [review]

**Comment by privacydied** [comment]
refactor(instagram): finalize regex-driven rewrite; remove redundant checks; add comprehensive tests

Per PR #444 review feedback:
- Remove unreachable kkinstagram.com guard (regex never matches it)
- Remove redundant IsValidInstagramHost() method (regex constrains hosts)
- Remove redundant IsAcceptedPath() method (regex restricts paths)
- Remove l.instagram.com support (link shim uses /?u= not /p/ or /reel/)
- Keep single debug log for successful rewrites

Added comprehensive unit tests:
- Positive: posts, reels, query/fragment preservation, trailing slashes
- Negative: already rewritten, non-supported endpoints, look-alike hosts
- Edge cases: case insensitivity, extra path segments, mixed content

Implementation:
- 40 lines total (down from 90)
- Regex-only gating for host and path
- Zero external calls, no changes to other handlers
- Preserves path, query, and fragment exactly

**Review by Sn0wCrack — APPROVED** [review]
Looks good now. Thanks for the contribution.

I'll give this a test myself before deploying as well.

**Comment by privacydied** [comment]
no problem dude enjoy
